crashtracker-ffi/src/collector_windows/api.rs

```diff
@@ -0,0 +1,737 @@
+}
+
+#[repr(C)]
+#[derive(Debug)]
+struct Guid {
+    pub data1: u32,
+    pub data2: u16,
+    pub data3: u16,
+    pub data4: [u8; 8],
+}
+
+impl fmt::LowerHex for Guid {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{:08x}{:04x}{:04x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}{:02x}",
+            self.data1,
+            self.data2,
+            self.data3,
+            self.data4[0],
+            self.data4[1],
+            self.data4[2],
+            self.data4[3],
+            self.data4[4],
+            self.data4[5],
+            self.data4[6],
+            self.data4[7]
+        )
+    }
+}
+
+#[repr(C)]
+struct CV_INFO_PDB70 {
+    pub signature: u32,
+    pub guid: Guid,
+    pub age: u32,
+}
+
+unsafe fn get_pdb_info(process_handle: HANDLE, base_address: u64) -> Result<PdbInfo> {
+    let dos_header: IMAGE_DOS_HEADER = read_memory(process_handle, base_address)?;
+
+    if dos_header.e_magic != IMAGE_DOS_SIGNATURE {
+        return Err(anyhow!("Invalid DOS header"));
+    }
+
+    let nt_headers_address = base_address + dos_header.e_lfanew as u64;
+    let nt_headers: IMAGE_NT_HEADERS_GENERIC = read_memory(process_handle, nt_headers_address)?;
+
+    if nt_headers.signature != IMAGE_NT_SIGNATURE {
+        return Err(anyhow!("Invalid NT headers"));
+    }
+
+    let is_pe64 = nt_headers.file_header.Machine == IMAGE_FILE_MACHINE_AMD64;
+    let is_pe32 = nt_headers.file_header.Machine == IMAGE_FILE_MACHINE_I386;
+
+    if !is_pe32 && !is_pe64 {
+        return Err(anyhow!("Invalid machine type"));
+    }
+
+    let debug_data_dir: IMAGE_DATA_DIRECTORY = if is_pe32 {
+        let nt_headers32: IMAGE_NT_HEADERS32 = read_memory(process_handle, nt_headers_address)?;
+        nt_headers32.OptionalHeader.DataDirectory[IMAGE_DIRECTORY_ENTRY_DEBUG.0 as usize]
+    } else {
+        let nt_headers64: IMAGE_NT_HEADERS64 = read_memory(process_handle, nt_headers_address)?;
+        nt_headers64.OptionalHeader.DataDirectory[IMAGE_DIRECTORY_ENTRY_DEBUG.0 as usize]
+    };
+
+    let debug_dir_buffer = read_memory_raw(
+        process_handle,
+        base_address + debug_data_dir.VirtualAddress as u64,
+        debug_data_dir.Size as usize,
+    )?;
+
+    let debug_dir = std::slice::from_raw_parts(
+        debug_dir_buffer.as_ptr() as *const IMAGE_DEBUG_DIRECTORY,
+        debug_dir_buffer.len() / std::mem::size_of::<IMAGE_DEBUG_DIRECTORY>(),
+    );
+
+    for entry in debug_dir {
+        if entry.Type != IMAGE_DEBUG_TYPE_CODEVIEW {
+            continue;
+        }
+
+        let cv_info: CV_INFO_PDB70 =
+            read_memory(process_handle, base_address + entry.AddressOfRawData as u64)?;
+
+        if cv_info.signature == 0x53445352
+        /* 'RSDS' */
+        {
+            return Ok(PdbInfo {
+                age: cv_info.age,
+                signature: cv_info.guid,
+            });
+        }
+    }
+
+    Err(anyhow!("No CodeView entry found"))
```

Comment:
anyhow::bail!

Reply:
Done in https://github.com/DataDog/libdatadog/pull/892/commits/ea8e7c31f38e2b3c86a21406ac52785ae29e7797